oxidation/libparsec/crates/types/src/addr.rs

```diff
@@ -26,7 +35,7 @@ macro_rules! impl_common_stuff {
             pub fn to_http_redirection_url(&self) -> Url {
                 let mut url = self.base.to_http_url(None);
                 url.path_segments_mut()
-                    .unwrap_or_else(|()| unreachable!())
+                    .expect("url is not a cannot-be-a-base")
```

Comment:
You use double negation can you rewrite this message ?

Reply:
no, because `cannot-be-a-base` is the specific term used in the URL RFC (https://docs.rs/reqwest/latest/reqwest/struct.Url.html#method.cannot_be_a_base)

Reply:
expect means the message describe why we expect it to be ok (cf. https://doc.rust-lang.org/std/result/enum.Result.html#recommended-message-style)

So here we expect the url not to be a cannot-be-a-base, otherwise [`Url::path_segments` returns `None`](https://docs.rs/url/latest/url/struct.Url.html#method.path_segments)

Reply:
I've changed the wording to make it more explicit what is expected